test/lib/BedrockTester.cpp

```diff
@@ -552,35 +559,66 @@ bool BedrockTester::waitForStates(set<string> states, uint64_t timeoutUS)
         }
         usleep(100'000);
     }
+
+    // we timed out, let's get some debugging info
+    string stateString;
+    auto it = states.begin();
+    while (it != states.end()) {
+        stateString += *it;
+        stateString += ",";
+        it++;
+    }
+    string statusString;
+    for (auto& item : json) {
+        statusString += item.first;
+        statusString += " ";
+        statusString += item.second;
+        statusString += " ";
+    }
+    cout << "waitForStates() timed out at " << STimeNow() << " waiting for states " << stateString << ". Started waiting at " << start << " Most recent status: " << statusString << endl;
+
     return false;
 }
 
-int BedrockTester::waitForPort(int port) {
-    int sock = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
-    int i = 1;
-    setsockopt(sock, SOL_SOCKET, SO_REUSEADDR, &i, sizeof(i));
-    sockaddr_in addr = {0};
-    addr.sin_family = AF_INET;
-    addr.sin_port = htons(port);
-    addr.sin_addr.s_addr = inet_addr("127.0.0.1");
-
-    int result = 0;
-    int count = 0;
+STable BedrockTester::getStatus(bool control) {
+    // FYI: sometimes the status command doesn't return with every key/value expected
+    return SParseJSONObject(executeWaitVerifyContent(SData("Status"), "200", control));
+}
+
+string BedrockTester::getStatusTerm(string term, bool control) {
+    // FYI: sometimes the status command doesn't return with every key/value expected
+    // Use with caution
+    return getStatus(control)[term];
+}
+
+bool BedrockTester::waitForStatusTerm(string term, string testValue, uint64_t timeoutUS, bool control){
     uint64_t start = STimeNow();
-    do {
-        result = ::bind(sock, (sockaddr*)&addr, sizeof(addr));
-        if (result) {
-            count++;
-            usleep(100'000);
-        } else {
-            shutdown(sock, 2);
-            close(sock);
-            return 0;
-        }
-    // Wait up to 30 seconds.
-    } while (result && STimeNow() < start + 30'000'000);
+    while (STimeNow() < start + timeoutUS) {
+        try {
+            string result = getStatusTerm(term, control);
 
-    return 1;
+            // if the value matches, return, otherwise wait
+            if (result != "" && result == testValue) {
```

Comment:
why the empty string check here? what if you want the test value to be an empty string?